### PR TITLE
Add a 404 page for the application

### PR DESCRIPTION
**Pull request by sewagodimo**
The 404 landing page, this makes user testing much simpler.
It's kinda awkward when students run unit a 404 Server Error

### CODE REVIEW

**Review by PiDelport — COMMENTED** [review]
We can probably do this without any custom view or such, and just a usual `404.html` template?

File: src/buza/templates/404.html [hunk 1]
```diff
@@ -0,0 +1,11 @@
+{% extends 'base.html' %}
+{% load static %}
+{% block title %}Page not found  404 {{ block.super }}{% endblock %}
+{% block head_extra %}
+    {{ block.super }}
+    <link rel="stylesheet" href="{% static 'buza/css/errors.css' %}">
```
Comment (PiDelport) [inline src/buza/templates/404.html]: @sewagodimo: This `errors.css` file seems to be missing: is this one you added locally, but haven't committed yet?

File: tests/test_views.py [hunk 2]
```diff
@@ -630,3 +631,16 @@ def test_get(self) -> None:
         self.assertContains(response, subject.title)
         self.assertContains(response, subject.description, count=1)
         self.assertContains(response, question.title, count=1)
+
+
+class Test404PageNotFound(TestCase):
```
Comment (sewagodimo) [inline tests/test_views.py:636]: @pjdelport I didn't know this, all the tutorials kept referring to it :)
Thank you the heads up

**Review by PiDelport — CHANGES_REQUESTED** [review]
@sewagodimo: Okay, I made the changes to switch this to using Django's built-in 404 view: all we need for that to work is to name our template `404.html`, without the prefix.

Can you look at that comment about `buza/css/errors.css`: is that something you still want to add?

File: src/buza/settings_env_dev.py [hunk 3]
```diff
@@ -27,4 +27,7 @@
     str(checkout_dir.joinpath('node_modules')),
 ]
 
+# Include the local host by default for development.
+ALLOWED_HOSTS = ['127.0.0.1', 'localhost']
```
Comment (sewagodimo) [inline src/buza/settings_env_dev.py:31]: @pjdelport do we still need this then?

**Comment by sewagodimo** [comment]
@pjdelport  I am happy if you merge, when travis is happy

File: src/buza/templates/404.html (hunk 1 above)
Comment (sewagodimo) [reply src/buza/templates/404.html]: No, I didn't realise that was still there
I want to style the 404 error page and the 400 error pages but it'snot a priority for now

File: src/buza/settings_env_dev.py (hunk 3 above)
Comment (PiDelport) [reply src/buza/settings_env_dev.py:31]: It's not a bad idea to keep: it's not used when `DEBUG` is enabled, but it's useful to have for testing / developing with `DEBUG` off (such as when testing the 404 template).

**Review by PiDelport — APPROVED** [review]

**Comment by PiDelport** [comment]
@sewagodimo: I think Travis got wedged somehow when I rebased and pushed this… I'm not sure how to get it unwedged. :/ Maybe it's because GitHub's current systems being partially down?

We can probably go ahead and merge it: the tests do pass.

I made a last tweak to the 404 page content: do you want to check it out and see if you're happy? If you are, go ahead and merge! :coffee:

**Comment by sewagodimo** [comment]
@pjdelport awesome, if I had known that you were free tonight I would have come to your place, next time!

File: src/buza/templates/404.html [hunk 4]
```diff
@@ -0,0 +1,11 @@
+{% extends 'base.html' %}
+{% load static %}
+{% block title %}Page not found  404 {{ block.super }}{% endblock %}
+
+{% block content %}
+    <div class="text-center">
+        <h1>We could not find the page you were looking for</h1>
+        <p style="font-size: 3rem">😭</p>
```
Comment (sewagodimo) [inline src/buza/templates/404.html:8]: super cute

**Comment by sewagodimo** [comment]
@pjdelport should we wait for the coverage report? It seems to be taking a while

**Comment by PiDelport** [comment]
@sewagodimo: Well, the branch coverage and patch coverage passes so far. I think the PR build (and its coverage) may just be stalled due to GitHub's technical issues.

We could wait until tomorrow (Tuesday), to see if it actually gets triggered and builds, and then merge if it does?

**Comment by PiDelport** [comment]
> @pjdelport awesome, if I had known that you were free tonight I would have come to your place, next time!

I was hacking from Codebridge tonight! :D

**Pull request merged**